### DISCUSSION TITLE
com.android.vending == com.android.vending /// as3 air????

### DISCUSSION THREAD

**Discussion by semoki** [airsdk/Adobe-Runtime-Support#2465] [Q&A]
In Unity, com.android.vending == com.android.vending

With this, you can check whether the file is received normally from Google Play.

In as3 air, no matter how much I google, I can't find the information, so I'm asking a question.

It seems like a basic function, but is there no such function?

**Comment by semoki** [comment]
public String getFromInstalledMarket(Context context, String targetPkg) {
	PackageManager pkgMgr = context.getPackageManager();
	return pkgMgr.getInstallerPackageName(targetPkg);
}

<----------In Unity

**Comment by marchbold** [comment]
Use the Package Manager ANE:

```as3
var installerPackageName:String = PackageManager.service.getInstallerPackageName();
if (installerPackageName == "com.android.vending") 
{
    // Installed via the Play Store
}
```

Available here: https://github.com/distriqt/ANE-PackageManager

or:

```
apm install com.distriqt.PackageManager
```

[Reference](https://distriqt.github.io/ANE-PackageManager/asdocs/com/distriqt/extension/packagemanager/PackageManager.html#getInstallerPackageName())

You can also get the install referrer: 

https://github.com/distriqt/ANE-AndroidSupport/wiki/com.android.installreferrer

**Reply by semoki** [reply]
After all, there was a way. Thank you very very much. Always be healthy ^____^
